tasks.py

```diff
@@ -62,43 +111,134 @@ def pip(ctx, command, **kwargs):
+    return poetry(ctx, f"run docgen {' '.join(flags)} {command}", **kwargs)
+
+
+def python_tool(ctx, tool, *args, **kwargs):
+    if tool[-3:] != ".py":
+        tool_path = TOOLS_DIR / f"{tool}.py"
+    else:
+        tool_path = TOOLS_DIR / tool
+    return poetry(ctx, f"run python {tool_path} {' '.join(args)}", **kwargs)
+
+
+def package_invoke(ctx, directory, command, **kwargs):
+    with ctx.cd(directory):
+        return _run(ctx, "invoke", command, **kwargs)
+
+
+def git(ctx, command, **kwargs):
+    return _run(ctx, "git", command, **kwargs)
+
+
+def invoke_each(ctx, command, **kwargs):
+    our_packages = _get_package_paths()
+    for package_path in our_packages.values():
+        return package_invoke(ctx, package_path, command, **kwargs)
+
+
+@task()
+def clean(ctx, venv=True, docs=False, all=False):
     """Cleans the virtual development environment by
     completely removing build artifacts and the .venv.
+    You can set ``--no-venv`` to avoid this default.
+
+    If ``--docs`` is supplied, the build artifacts for
+    local documentation will also be cleaned.
+
+    You can set flag ``all`` to clean all packages as
+    well.
     """
-    for pattern in CLEAN_PATTERNS:
+    union_clean_patterns = []
+    if venv:
+        union_clean_patterns.extend(CLEAN_PATTERNS)
+    if docs:
+        union_clean_patterns.extend(DOCS_CLEAN_PATTERNS)
+    for pattern in union_clean_patterns:
         for path in glob(pattern, recursive=True):
             print(f"Removing: {path}")
             shutil.rmtree(path, ignore_errors=True)
+            try:
+                os.remove(path)
+            except OSError:
+                pass
+    if all:
+        invoke_each(ctx, "clean")
+
+
+@task
+def setup_poetry(ctx, username=None, password=None, token=None):
+    """Configure local poetry installation for development.
+    If you provide ``username`` and ``password``, you can
+    also configure your pypi access. Our version of poetry
+    uses ``keyring`` so the password is not stored in the
+    clear.
+
+    Alternatively, you can set ``token`` to use a pypi token, be sure
+    to include the ``pypi-`` prefix in the token.
+    """
+    poetry(ctx, "config -n --local virtualenvs.in-project true")
+    poetry(ctx, "config -n --local virtualenvs.create true")
+    poetry(ctx, "config -n --local virtualenvs.path null")
+    poetry(ctx, "config -n --local experimental.new-installer true")
+    poetry(ctx, "config -n --local installer.parallel true")
+    poetry(
+        ctx,
+        "config -n --local repositories.devpi.url 'https://devpi.robocorp.cloud/ci/test'",
+    )
+    if username and password and token:
+        raise ParseError(
+            "You cannot specify username-password combination and token simultaneously"
+        )
+    if username ^ password:
+        raise ParseError("You must specify both username and password")
+    if username and password:
+        poetry(ctx, f"config -n http-basic.pypi {username} {password}")
```

Comment:
You sure this works? (fails on my side with: `TypeError: unsupported operand type(s) for ^: 'NoneType' and 'NoneType'`)

What about:
```suggestion
    if username and password:
        poetry(ctx, f"config -n http-basic.pypi {username} {password}")
    else:
        raise ParseError("You must specify both username and password")        
```